app/assets/javascripts/admin/addon/components/schema-theme-setting/types/categories.gjs

```diff
@@ -0,0 +1,73 @@
+import Component from "@glimmer/component";
+import { tracked } from "@glimmer/tracking";
+import { action } from "@ember/object";
+import { and, not } from "truth-helpers";
+import I18n from "discourse-i18n";
+import FieldInputDescription from "admin/components/schema-theme-setting/field-input-description";
+import CategorySelector from "select-kit/components/category-selector";
+
+export default class SchemaThemeSettingTypeCategories extends Component {
+  @tracked touched = false;
+
+  @tracked
+  value =
+    this.args.value?.map((categoryId) => {
+      return this.args.setting.metadata.categories[categoryId];
```

Comment:
This is not very clear to me, but it looks like `this.args.setting.metadata.categories` contains just the previously selected categories, right?

Reply:
Yes this is correct because the dropdown requires category objects to be passed to it.